frontend/src/component/common/Limit/Limit.tsx

```diff
@@ -0,0 +1,144 @@
+    marginBottom: theme.spacing(2),
+}));
+
+const LimitExplanation = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginTop: theme.spacing(1.5),
+}));
+
+const ExpandableBox = styled(Box)(({ theme }) => ({
+    flex: 1,
+}));
+
+export const Limit: FC<{
+    name: string;
+    shortName?: string;
+    limit: number;
+    currentValue: number;
+    onClose?: () => void;
+}> = ({ name, shortName, limit, currentValue, onClose }) => {
+    const percentageLimit = Math.round((currentValue / limit) * 100);
+    const belowLimit = currentValue < limit;
+    const threshold = 80;
+
+    if (percentageLimit < threshold) {
+        return null;
+    }
+
+    return (
+        <StyledBox>
+            <Header>
+                <ConditionallyRender
+                    condition={belowLimit}
+                    show={<StyledWarningIcon />}
+                    elseShow={<StyledErrorIcon />}
+                />
+
+                <ConditionallyRender
+                    condition={belowLimit}
+                    show={
+                        <ExpandableBox>
+                            You are nearing the limit for {name}
+                        </ExpandableBox>
+                    }
+                    elseShow={
+                        <ExpandableBox>
+                            You reached the limit for {name}
+                        </ExpandableBox>
+                    }
+                />
+
+                <ConditionallyRender
+                    condition={typeof onClose === 'function'}
+                    show={
+                        <Tooltip title='Close' arrow describeChild>
+                            <IconButton onClick={onClose}>
+                                <CloseIcon />
+                            </IconButton>
+                        </Tooltip>
+                    }
+                />
+            </Header>
+            <Main>
+                <LimitStats>
+                    You have added {currentValue} {shortName ?? name}, that is
+                    equivalent to{' '}
+                    <Typography component='span' color='primary'>
+                        {percentageLimit}%
+                    </Typography>{' '}
+                    of the limit.
+                </LimitStats>
+                <BorderLinearProgress
+                    variant='determinate'
+                    value={Math.min(100, percentageLimit)}
+                />
+                <LimitExplanation>
+                    <Link
+                        target='_blank'
+                        to={'https://docs.getunleash.io/reference/limits'}
+                    >
+                        Read more about limits
+                    </Link>
+                    <Typography fontWeight='bold'>Limit: {limit}</Typography>
+                </LimitExplanation>
+            </Main>
+            <Footer>
+                If you need more than <b>{limit}</b> {shortName ?? name}, please
```

Comment:
These should probably be `<strong>` instead. According to mdn:
> you should not use `<b>` for styling text or granting importance. If you wish to create boldface text, you should use the CSS [font-weight](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight) property. If you wish to indicate an element is of special importance, you should use the [`<strong>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong) element.

```suggestion
                If you need more than <strong>{limit}</strong> {shortName ?? name}, please
```